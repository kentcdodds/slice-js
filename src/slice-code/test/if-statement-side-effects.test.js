import {runAllCombosTests} from './helpers/utils'

// fit(
//   'ifWithAssignment(false)',
//   require('./helpers/utils').snapSlice(require.resolve('./fixtures/if-statement-side-effects'), ({ifWithAssignment}) => {
//     ifWithAssignment(false)
//   })
// )


runAllCombosTests({
  filename: require.resolve('./fixtures/if-statement-side-effects'),
  methods: [
    {methodName: 'ifWithAssignment', possibleArguments: [[true], [false]]},
    {methodName: 'ifWithFunctionCall', possibleArguments: [[true], [false]]},
  ],
})
