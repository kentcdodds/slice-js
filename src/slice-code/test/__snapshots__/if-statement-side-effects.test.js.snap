exports[`test ifWithAssignment(false) 1`] = `
"export { ifWithAssignment };

function ifWithAssignment(passIf) {
  let x = 'hi';

  return !passIf + x;
}"
`;

exports[`test ifWithAssignment(true) && ifWithAssignment(false) 1`] = `
"export { ifWithAssignment };

function ifWithAssignment(passIf) {
  let x;
  if ((x = 'hi') && passIf) {
    return x + passIf;
  } else {
    return !passIf + x;
  }
}"
`;

exports[`test ifWithAssignment(true) 1`] = `
"export { ifWithAssignment };

function ifWithAssignment(passIf) {
  let x = 'hi';

  return x + passIf;
}"
`;

exports[`test ifWithFunctionCall(false) 1`] = `
"export { ifWithFunctionCall };

function ifWithFunctionCall(passIf) {
  let thing;

  (a => thing = a)('hey');
  return !passIf;
}"
`;

exports[`test ifWithFunctionCall(true) && ifWithFunctionCall(false) 1`] = `
"export { ifWithFunctionCall };

function ifWithFunctionCall(passIf) {
  let thing;

  if (passIf && (a => thing = a)('hey')) {
    return thing + passIf;
  } else {
    return !passIf;
  }
}"
`;

exports[`test ifWithFunctionCall(true) 1`] = `
"export { ifWithFunctionCall };

function ifWithFunctionCall(passIf) {
  let thing;

  (a => thing = a)('hey');
  return thing + passIf;
}"
`;
